common/src/main/java/com/wynntils/core/managers/ManagerRegistry.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.managers;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.core.commands.ClientCommandManager;
+import com.wynntils.core.config.ConfigManager;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.overlays.OverlayManager;
+import com.wynntils.core.keybinds.KeyManager;
+import com.wynntils.core.webapi.WebManager;
+import com.wynntils.mc.utils.CompassManager;
+import com.wynntils.mc.utils.CrashReportManager;
+import com.wynntils.wc.ModelManager;
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.apache.commons.lang3.reflect.MethodUtils;
+
+public class ManagerRegistry {
+    private static final List<Class<? extends Manager>> PERSISTENT_MANAGERS = new ArrayList<>();
+    private static final Map<Class<? extends Manager>, List<Feature>> MANAGER_DEPENDENCIES = new HashMap<>();
+    private static final Set<Class<? extends Manager>> ENABLED_MANAGERS = new HashSet<>();
+
+    public static void init() {
+        registerPersistentDependency(CompassManager.class);
```

Comment:
This looks like alphabetic order. Did the bootstrap order issue resolve itself with this patch?

Reply:
The bootstrap issue is not present here because persistent managers are loaded before features and their dependency manager. Order does not matter with persistent managers.

Reply:
So basically the ordering problem was "first persistent managers (in any order), then feature managers (in any order)"?

Reply:
There might be cases where it will not be true (Web configs will need WebManager for configs). At the moment, this works fine.

Reply:
Is CompassManager really essential? I thought it was just used by features like beacon tracking.

Reply:
It is used by territory command. I guess I can make it not persistent, but then we will have no use case it for now.

Reply:
Done